backend/de.metas.dunning/src/main/java/de/metas/dunning/document/archive/DunningDocOutboundLogMailRecipientProvider.java

```diff
@@ -105,6 +109,43 @@ public Optional<DocOutBoundRecipient> provideMailRecipient(@NonNull final DocOut
 
 		final String locationEmail = dunningService.getLocationEmail(dunningDocId);
 
+
+		if (dunningContact!= null  && dunningContact.isDunningContact())
+		{
+			final DocOutBoundRecipient dunningRecipient = recipientRepository.getById(DocOutBoundRecipientId.ofRepoId(dunningContact.getAD_User_ID()));
+
+			final String dunningEmail = dunningContact.getEMail();
+
+
+			if (Check.isNotBlank(dunningEmail))
+			{
+				Loggables.addLog("The dunned invoices all have dunning user ={} and the dunning contact has the email {} so we take that user as the dunning mail's participant, with this email address", dunningRecipient, dunningEmail);
+				return Optional.of(dunningRecipient.withEmailAddress(dunningEmail));
+			}
+
+			if (Check.isNotBlank(invoiceEmail))
+			{
+				Loggables.addLog("The dunned invoices all have invoiceUser={} and the invoice has the email {} so we take that user as the dunning mail's participant, with this email address", dunningRecipient, invoiceEmail);
+				return Optional.of(dunningRecipient.withEmailAddress(invoiceEmail));
+			}
+
+			if (Check.isNotBlank(dunningRecipient.getEmailAddress()))
+			{
+				Loggables.addLog("The dunned invoices all have invoiceUser={}, so we take that user as the dunning mail's participant", dunningRecipient);
+				return Optional.of(dunningRecipient);
+			}
+
+			if (Check.isNotBlank(locationEmail))
+			{
+				Loggables.addLog("The dunned invoices all have invoiceUser={} and the location has the email {} so we take that user as the dunning mail's participant, with this email address", dunningRecipient, locationEmail);
+				return Optional.of(dunningRecipient.withEmailAddress(invoiceEmail));
```

Comment:
```suggestion
				return Optional.of(dunningRecipient.withEmailAddress(locationEmail));
```
Shouldn't this be using `locationEmail`, which is what we're checking as not-null? If `invoiceEmail` would be usable, it should have been used already 2 ifs above this one

Reply:
Yes, right!
Was a bug in the code below, that I did not notice, and I copied here as well.
Fixed here and in the bellow if.